src/ResponseProcessor/SimpleXmlResponseProcessor.php

```diff
@@ -94,17 +94,20 @@ protected function logBadXML($xml)
     /**
      * Sanitize XML.
      *
-     * @param string $xml XML to sanitize
+     * @param string $rawXml XML to sanitize
      *
      * @return string
      */
-    protected function sanitizeXml($xml)
+    protected function sanitizeXml($rawXml)
     {
+        // Make sure the encoding is correct before applying regular expressions:
+        $xml = mb_convert_encoding($rawXml, 'UTF-8', 'UTF-8');
+
         // Sanitize the XML if requested:
         $newXML = trim(preg_replace($this->sanitizeRegex, ' ', $xml, -1, $count));
 
-        if ($count > 0 && $this->badXmlLog) {
-            $this->logBadXML($xml);
+        if (($xml !== $rawXml || $count > 0) && $this->badXmlLog) {
```

Comment:
Shouldn't this comparison be between `$newXml` and `$rawXml`?

Reply:
That was actually intentional, though maybe I'm overthinking it. My idea was that `$count` tells us if something changed during the preg_replace step, but we also want to know if something changed as a result of the UTF-8 normalization... so I checked those two things independently to ensure that if either causes a change, it gets logged. Does that make any sense? I'm happy to change it if that makes things more clear, but I'm not sure that it makes a difference to the end result. (Another related question: is there any reason to check count at all if we're just comparing the initial input with the final output? Maybe that's the simplest approach of all).

Reply:
And you're right, the current code works fine, but is more complex than necessary.